python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -242,10 +253,8 @@ def runQuantum(self, butlerQC: pipeBase.ButlerQuantumContext,
                    inputRefs: pipeBase.InputQuantizedConnection,
                    outputRefs: pipeBase.OutputQuantizedConnection):
         inputs = butlerQC.get(inputRefs)
-        expId, expBits = butlerQC.quantum.dataId.pack("visit_detector",
-                                                      returnMaxBits=True)
-        idFactory = self.makeIdFactory(expId=expId, expBits=expBits)
-
+        idGenerator = self.config.idGenerator.apply(butlerQC.quantum.dataId)
+        idFactory = idGenerator.make_table_id_factory()
```

Comment:
Should these be moved into `run`? I've wondered about why we do it in `runQuantum` some of the time.

Reply:
Answering my own question after looking at the pipe_tasks code: because we need the dataId, which isn't available in `run`. See my other comments in pipe_tasks, though.